src/hotspot/share/opto/compile.cpp

```diff
@@ -4436,8 +4450,21 @@ void Compile::remove_speculative_types(PhaseIterGVN &igvn) {
   }
 }
 
-// Auxiliary method to support randomized stressing/fuzzing.
-//
+// Auxiliary methods to support randomized stressing/fuzzing.
+
+int Compile::random() {
+  _stress_seed = os::next_random(_stress_seed);
+  return static_cast<int>(_stress_seed);
+}
+
+void Compile::shuffle(Unique_Node_List* l) {
+  if (l->size() < 2) return;
+  for (uint i = l->size() - 1; i >= 1; i--) {
```

Comment:
You can remove the if-check as the loop check already covers it (loop is only executed if size >= 2).

Reply:
Note that `size()` could be 0, leading `l->size() - 1` to underflow.

Reply:
You're right, it's `uint` and not `int` - then you can leave it as it is.